Update to latest vuetify

scottwittenburg opened the pull request:
Rather than exhaustively going through the [migration guide](https://vuetifyjs.com/en/getting-started/releases-and-migrations/#migration-guide) looking for usages in the application, I just tried to exercise all of the app (that is enabled for me, I can't actually send an email for example) to discover issues, then used the guide to figure out how to fix them.

- [x] Fix problems with keyboard shortcuts dialog
- [x] Fix deprecation warning and lock icon alignment in girder user modal dialog
- [x] Fix alignment of window/level sliders with numeric inputs
- [x] Fix dependency issue that arose since yesterday.  The problem is a failure to compile some `vue-async-computed` features from Girder web components, `npm install` on yesterdays commit results in breakage, where `npm ci` on the same commit works fine.

scottwittenburg commented:
I'm not sure what we need to do to move forward to the most recent release of the @babel packages I pinned in [0ea1621](https://github.com/OpenImaging/miqa/pull/21/commits/0ea1621c7ac4662c76fa26021e21bd2fa1f4afab).  But with [7.11.1](https://github.com/babel/babel/releases/tag/v7.11.1), I started getting compilation failures like:

```
 ERROR  Failed to compile with 5 errors                                                                                                                                                                                              6:44:47 PM

 error  in ./node_modules/@girder/components/src/components/Breadcrumb.vue?vue&type=script&lang=js&

Syntax Error: SyntaxError: /Users/scott/projects/miqa/miqa/client/node_modules/@girder/components/src/components/Breadcrumb.vue: Unexpected token (40:15)

  38 |     pathBreadcrumb: {
  39 |       default: [],
> 40 |       async get() {
     |                ^
  41 |         this.loading = true;
  42 |         const breadcrumb = [];
  43 |         const { rootLocationDisabled, girderRest, location } = this;


 @ ./node_modules/@girder/components/src/components/Breadcrumb.vue?vue&type=script&lang=js& 1:0-251 1:267-270 1:272-520 1:272-520
 @ ./node_modules/@girder/components/src/components/Breadcrumb.vue
 @ ./node_modules/@girder/components/src/components/index.js
 @ ./node_modules/@girder/components/src/index.js
 @ ./src/main.js
 @ multi (webpack)-dev-server/client?http://localhost:8081/sockjs-node (webpack)/hot/dev-server.js ./src/main.js
```

Sticking with the previous release version, `7.11.0` works around the problem for the time being.

scottwittenburg commented:
fyi @aashish24

aashish24 commented:
@scottwittenburg I am thinking if we can have someone perhaps @jourdain brandon davis or @bryonlewis review our work in MIQA.

scottwittenburg commented on client/src/components/KeyboardShortcutDialog.vue:
```diff
@@ -50,16 +50,24 @@ export default {
+          hide-default-footer
+          hide-default-header
+        >
+          <template v-slot:body="{ items }">
```
Try to use item slot rather than body slot

scottwittenburg commented on client/src/components/KeyboardShortcutDialog.vue:
```diff
@@ -50,16 +50,24 @@ export default {
+            <tbody>
+              <tr v-for="item in items" :key="item[0]">
+                <td>{{ item[0] }}</td>
+                <td v-html="item[1]"></td>
```
Factor out so we don't have the repeated `<code></code>` text in the data above.

scottwittenburg commented on client/src/components/KeyboardShortcutDialog.vue:
```diff
@@ -50,16 +50,24 @@ export default {
+        >
+          <template v-slot:body="{ items }">
+            <tbody>
+              <tr v-for="item in items" :key="item[0]">
```
Use the index for the key `for (item,i)` instead of the name, which should be safe since this is a static list.  Not relevant if we use the item slot instead of body.

scottwittenburg commented:
Just curious if you think you'll have a chance to look at this @BryonLewis?

BryonLewis commented:
I should be able to take a look at it tomorrow afternoon.  Could you assign me as a reviewer?

scottwittenburg commented:
> I should be able to take a look at it tomorrow afternoon. Could you assign me as a reviewer?

Thanks!  Although I typed your github username in as a reviewer, it insists there is "nothing to show", and the list of reviewers remains unchanged.  I'll keep trying.

jourdain commented:
That means you need to add him to the project before you can add him as a reviewer.

scottwittenburg commented on client/src/components/CSVConfig.vue:
```diff
@@ -51,7 +51,7 @@ export default {
           v-model="importpath"
           @input="changed = true"
-          browser-autocomplete="on"
+          autocomplete="on"
```
Thanks for providing some useful background info on this, I'll just resolve for now.

scottwittenburg commented on client/src/components/SiteConfig.vue:
```diff
@@ -57,7 +57,7 @@ export default {
     :headers="headers"
     :items="items"
-    hide-actions
+    hide-default-filter
```
Yep, good catch, thanks.  According to the [migration guide](https://vuetifyjs.com/en/getting-started/releases-and-migrations/#releases-and-migrations), `hide-actions` has been renamed to `hide-default-footer`, not `hide-default-filter`.  I just mistyped that somehow, can't even blame sublime for auto-completing incorrectly in this case.

scottwittenburg commented on client/src/views/Dataset.vue:
```diff
@@ -522,20 +518,23 @@ export default {
-                    </v-btn>
-                    <span>Revert</span>
-                  </v-tooltip>
+                  <template v-slot:activator="{ on }">
+                    <v-tooltip top>
+                      <v-btn
+                        slot="activator"
+                        text
+                        icon
+                        small
+                        color="grey"
+                        class="my-0"
+                        v-on="on"
+                        @click="loadSessionMeta"
+                      >
+                        <v-icon>undo</v-icon>
+                      </v-btn>
+                      <span>Revert</span>
+                    </v-tooltip>
+                  </template>
```
Yes, with your suggested change, I could see that button and tooltip, once I found data that passed the condition for rendering it.  Thanks!

scottwittenburg commented:
Thanks for reviewing this @BryonLewis 😃 I think I addressed your concerns, can you take another look when you get a chance, and approve if it looks ok to you?

scottwittenburg commented:
> Looks good based on the code changes.

I think that's all we can really ask of you, thanks for taking the time to have a look and provide some feedback!
